scripts/0trace-test/modify-websocket.go

```diff
@@ -0,0 +1,184 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+	"github.com/gorilla/websocket"
+	"html/template"
+	"log"
+	"net"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+var buffer gopacket.SerializeBuffer
+var options gopacket.SerializeOptions
+
+const TCPTimeout = time.Duration(1000) * time.Millisecond
+
+var upgrader = websocket.Upgrader{} // use default options
+
+func tcpConn(clientIP string, clientPort string, netConn net.Conn) {
+	tcpConn := netConn.(*net.TCPConn)
+
+	clPort, _ := strconv.Atoi(clientPort)
+	dstIP := net.ParseIP(clientIP)
+	// Send raw bytes over wire
+	rawBytes := []byte("heeloo tcp")
+
+	ipLayer := &layers.IPv4{
+		Protocol: layers.IPProtocolTCP,
+		Version:  4,
+		DstIP:    dstIP,
+		TTL:      uint8(5), // low TTL so it does not reach client
+	}
+	tcpLayer := &layers.TCP{
+		SrcPort: layers.TCPPort(443),
+		DstPort: layers.TCPPort(clPort),
+		Seq:     11111,
+		ACK:     true,
+		PSH:     true,
+	}
+	// And create the packet with the layers
+	buffer = gopacket.NewSerializeBuffer()
+	gopacket.SerializeLayers(buffer, options,
+		ipLayer,
+		tcpLayer,
+		gopacket.Payload(rawBytes),
+	)
+	outgoingPacket := buffer.Bytes()
+
+	n, err := tcpConn.Write([]byte(outgoingPacket))
+	if err != nil {
+		log.Print(err)
```

Comment:
This error seems worth propagating to me, i.e., return it back to the caller, which is the echo handler. Also, if we fail to write the packets to the wire, is there even a point in continuing execution in the echo handler?